### PR TITLE
when assigning delete groups to add

### PR DESCRIPTION
**Pull request by xacobofg**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #6906

### CODE REVIEW

File: inc/ruleticket.class.php [hunk 1]
```diff
@@ -230,6 +230,10 @@ function executeActions($output, $params, array $input = []) {
                            "items_id" => $output['id']]);
                      }
                   }
+                  
+                  if ($action->fields["field"]=='_groups_id_assign' && array_key_exists('_additional_groups_assigns', $output)) {
+                     unset($output['_additional_groups_assigns']);
+                  }
```
Comment (cedric-anne) [inline inc/ruleticket.class.php:236]: Proposed behaviour seems correct (I would like a confirmation from someone; ping @flegastelois @orthagh ), but I think the fix should be global (i.e., behaviour should be the same for users / groups / suppliers and for requesters / observers / assigned).

Also, if proposal is accepted, we have to add some tests on this.

```suggestion
                  $actions = $this->getActions();
                  $append_key = $actions[$action->fields["field"]]["appendto"] ?? null;
                  if ($append_key !== null && array_key_exists($append_key, $output)) {
                     unset($output[$append_key]);
                  }
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Sorry for the late reply.

File: inc/ruleticket.class.php (hunk 1 above)
Comment (orthagh) [reply inc/ruleticket.class.php:236]: > I would like a confirmation from someone; ping @flegastelois @orthagh

I agree

> behaviour should be the same for users / groups / suppliers and for requesters / observers / assigned

also agree, for actions where you can choose between appending or assigning

File: inc/ruleticket.class.php [hunk 2]
```diff
@@ -230,6 +230,12 @@ function executeActions($output, $params, array $input = []) {
                            "items_id" => $output['id']]);
                      }
                   }
+
```
Comment (cedric-anne) [inline inc/ruleticket.class.php:233]:
```suggestion

```

**Comment by cedric-anne** [comment]
Code seems ok, but some unit tests have to be added to validate it.

**Comment by cedric-anne** [comment]
Hi @xacobofg ,

Can you add some unit tests on this PR ?

Regards

File: tests/functionnal/RuleTicket.php [hunk 3]
```diff
@@ -480,4 +480,94 @@ public function testITILSolutionAssignFromRule() {
...
+         'pattern'   => 1,
+      ]);
+      $this->checkInput($rulecrit, $crit_id, $crit_input);
+
+      //create action to add group as group requester
+      $action_id = $ruleaction->add($action_input = [
+         'rules_id'    => $ruletid,
+         'action_type' => 'add',
+         'field'       => '_groups_id_requester',
+         'value'       => 1,
```
Comment (cedric-anne) [inline tests/functionnal/RuleTicket.php:516]: You should create groups first, in order to use `$group1` value here.

**Review by cedric-anne — COMMENTED** [review]
Thank you for responding to my request to add tests.

I have some comments.

1. You have to resolve conflicts to ensure test suite is executed.
2. Having a rule that have both "add" and "assign" action on the same field seems weird. Maybe using 2 different rules on the test would fit more to a common usage.

**Review by cedric-anne — APPROVED** [review]

**Comment by cedric-anne** [comment]
I've rebase and squash commits, to fix some conflicts.

**Review by trasher — APPROVED** [review]

File: inc/ruleticket.class.php [hunk 4]
```diff
@@ -238,6 +238,12 @@ function executeActions($output, $params, array $input = []) {
                      }
                   }
 
+                  $actions = $this->getActions();
```
Comment (cedric-anne) [inline inc/ruleticket.class.php:241]:
```suggestion
                  // Remove values that may have been added by any "append" rule action on same field.
                  $actions = $this->getActions();
```

**Review by orthagh — APPROVED** [review]

**Comment by cedric-anne** [comment]
Thanks for the contribution.

**Pull request merged**